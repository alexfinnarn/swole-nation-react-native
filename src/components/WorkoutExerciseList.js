import React from 'react';
import {FlatList, View} from "react-native";
import {styles} from "./Styles";
import WorkoutExerciseListItem from "./WorkoutExerciseListItem";

export default function WorkoutExerciseList({ exercises, workout, handleSelection }) {

  // Add key that is needed for rendering the <FlatList /> component.
  exercises = exercises.map((exercise) => {
    exercise.key = exercise.id.toString();
    return exercise;
  });

  return (
    <View>
      <FlatList
        data={exercises}
        renderItem={(item) => <WorkoutExerciseListItem exercise={item} updater={handleSelection} workout={workout}/>}
      />
    </View>
  )
}
