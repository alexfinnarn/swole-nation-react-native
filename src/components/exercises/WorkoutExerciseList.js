import React from 'react';
import {FlatList, View} from "react-native";
import WorkoutExerciseListItem from "./WorkoutExerciseListItem";

export default function WorkoutExerciseList({ exercises, workout, handleSelection, navigation }) {
  // Add key that is needed for rendering the <FlatList /> component.
  exercises = exercises.map((exercise) => {
    exercise.key = exercise.id.toString();
    return exercise;
  });

  return (
    <View>
      <FlatList
        data={exercises}
        renderItem={(item) => <WorkoutExerciseListItem navigation={navigation} exercise={item} updater={handleSelection} workout={workout}/>}
      />
    </View>
  )
}
