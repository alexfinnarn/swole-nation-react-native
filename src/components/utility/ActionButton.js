import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {home} from "../Styles";

export default function ActionButton({text, action, label, styles, disabled = false}) {
  const bgColor = disabled ? '#bababa' : '#21897E';

  return (
    <View style={[{flex: 1, flexDirection: 'column'}, styles]}>
      <TouchableOpacity
        disabled={disabled}
        accessible={true}
        accessibilityLabel={label}
        activeOpacity={0.6}
        style={[{backgroundColor: bgColor}, home.actionButton]}
        onPress={action}>
        <View style={{flex: 1, justifyContent: 'center'}}>
          <Text style={home.actionButtonText}>{text}</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
