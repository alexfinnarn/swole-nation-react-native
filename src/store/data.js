import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
