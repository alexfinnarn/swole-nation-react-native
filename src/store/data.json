{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
