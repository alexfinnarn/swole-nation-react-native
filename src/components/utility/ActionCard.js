import React from 'react';
import {Text, View} from 'react-native';
import {home} from '../Styles';
import ActionButton from './ActionButton';

export default function ActionCard({children, actionComponent}) {
  return (
    <View style={home.sectionContainer}>
      <View style={home.sectionLeft}>{children}</View>
      {actionComponent}
    </View>
  );
}
