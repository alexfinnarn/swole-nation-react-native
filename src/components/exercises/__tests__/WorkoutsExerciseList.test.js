import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import data from "../../../store/data";
import WorkoutExerciseList from "../WorkoutExerciseList";
import ActionCard from "../../utility/ActionCard";
import {TouchableOpacity} from "react-native-web";
import ActionButton from "../../utility/ActionButton";

const strongliftsA = 'DQkECwYLCQQ';
const strongliftsB = 'DAwJBQEMAA0';
const handle = {
  createExercise: jest.fn(),
  setActiveExerciseKey: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};
let renderer, instance ={};
let exercises = {};

function getExercises(key) {
  return data.workouts[key].exercises.map((name) => {
    const foundKey = Object.keys(data.exercises).find(key => data.exercises[key].name === name);
    return data.exercises[foundKey];
  });
}


beforeEach(() => {
  exercises = getExercises(strongliftsA);
  renderer = render(<WorkoutExerciseList exercises={exercises} navigation={navigation} handle={handle} />);
  instance = renderer.getByTestId('workout-exercise-list-root');
}, 0);

describe('<WorkoutExerciseList />', () => {
  it('Renders correctly', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('Renders the correct amount of items per workout', () => {
    let actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(6);

    expect(renderer.queryByText('Deadlifts')).toBeNull();
    expect(renderer.queryByText('Bench Press')).not.toBeNull();

    exercises = getExercises(strongliftsB);
    renderer.update(<WorkoutExerciseList exercises={exercises} navigation={navigation} handle={handle} />)

    actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(6);

    expect(renderer.queryByText('Deadlifts')).not.toBeNull();
    expect(renderer.queryByText('Bench Press')).toBeNull();

  });

  it('Goes to correct exercise when Edit is pressed', () => {
    let editButtons = renderer.getAllByText('Edit');

    // Edit Squats Warmup.
    fireEvent(editButtons[0], 'press');
    expect(handle.setActiveExerciseKey).toHaveBeenCalledTimes(1);
    expect(handle.setActiveExerciseKey).toHaveBeenCalledWith('BwEBDwoFBgQ');

    // Edit Barbell Row.
    fireEvent(editButtons[editButtons.length - 1], 'press');
    expect(handle.setActiveExerciseKey).toHaveBeenCalledTimes(2);
    expect(handle.setActiveExerciseKey).toHaveBeenLastCalledWith('BwQKAwYKAgA');
  });
});
