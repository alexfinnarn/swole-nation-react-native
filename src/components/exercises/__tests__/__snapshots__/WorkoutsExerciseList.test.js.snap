// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutExerciseList /> Renders correctly 1`] = `
<View
  testID="workout-exercise-list-root"
>
  <RCTScrollView
    data={
      Array [
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "BwEBDwoFBgQ",
          "name": "Squats Warmup",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 45,
            },
            Object {
              "reps": 5,
              "weight": 45,
            },
            Object {
              "reps": 5,
              "weight": 95,
            },
            Object {
              "reps": 5,
              "weight": 135,
            },
            Object {
              "reps": 3,
              "weight": 185,
            },
            Object {
              "reps": 2,
              "weight": 205,
            },
          ],
        },
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "Bw0ECwAKBws",
          "name": "Squats",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 230,
            },
            Object {
              "reps": 5,
              "weight": 230,
            },
            Object {
              "reps": 5,
              "weight": 230,
            },
            Object {
              "reps": 5,
              "weight": 230,
            },
            Object {
              "reps": 5,
              "weight": 230,
            },
          ],
        },
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "BwwBDAIJDgw",
          "name": "Bench Press Warmup",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 45,
            },
            Object {
              "reps": 5,
              "weight": 45,
            },
            Object {
              "reps": 5,
              "weight": 95,
            },
            Object {
              "reps": 3,
              "weight": 135,
            },
          ],
        },
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "DgcBCA8BCQo",
          "name": "Bench Press",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 165,
            },
            Object {
              "reps": 5,
              "weight": 165,
            },
            Object {
              "reps": 5,
              "weight": 165,
            },
            Object {
              "reps": 5,
              "weight": 165,
            },
            Object {
              "reps": 5,
              "weight": 165,
            },
          ],
        },
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "BwYGAw0NDwo",
          "name": "Barbell Row Warmup",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 95,
            },
          ],
        },
        Object {
          "image": 1,
          "instructions": "Do with a barbell",
          "key": "BwQKAwYKAgA",
          "name": "Barbell Row",
          "sets": Array [
            Object {
              "reps": 5,
              "weight": 115,
            },
            Object {
              "reps": 5,
              "weight": 115,
            },
          ],
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Squats Warmup
            </Text>
            <Text>
              5x45 lbs, 
              5x45 lbs, 
              5x95 lbs, 
              5x135 lbs, 
              3x185 lbs, 
              2x205 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Squats
            </Text>
            <Text>
              5x230 lbs, 
              5x230 lbs, 
              5x230 lbs, 
              5x230 lbs, 
              5x230 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Bench Press Warmup
            </Text>
            <Text>
              5x45 lbs, 
              5x45 lbs, 
              5x95 lbs, 
              3x135 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Bench Press
            </Text>
            <Text>
              5x165 lbs, 
              5x165 lbs, 
              5x165 lbs, 
              5x165 lbs, 
              5x165 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Barbell Row Warmup
            </Text>
            <Text>
              5x95 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "elevation": 5,
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 4,
              "marginTop": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 4,
                "flexDirection": "column",
                "padding": 10,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "flex": 4,
                    "paddingBottom": 10,
                  },
                ]
              }
            >
              Barbell Row
            </Text>
            <Text>
              5x115 lbs, 
              5x115 lbs
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#21897E",
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
